internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalCommon.scala

```diff
@@ -382,10 +382,10 @@ private[inc] abstract class IncrementalCommon(
       log.debug(s"current source = $sources")
 
       new UnderlyingChanges[VirtualFileRef] {
-        val added0 = new java.util.HashSet[VirtualFileRef]
-        val changed0 = new java.util.HashSet[VirtualFileRef]
-        val removed0 = new java.util.HashSet[VirtualFileRef]
-        val unmodified0 = new java.util.HashSet[VirtualFileRef]
+        val added0 = java.util.concurrent.ConcurrentHashMap.newKeySet[VirtualFileRef]
```

Comment:
Thanks for addressing this! I'm not going to review the PR because I have no context on the other commit.

Reply:
I'm going to split that into a separate PR - https://github.com/sbt/zinc/pull/850